### PR TITLE
Add translation Api and use it in TinyMce for custom buttons

### PR DESCRIPTION
**Pull request by itkg-ngilain**
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1073
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/38

### CODE REVIEW

**Comment by itkg-alavieille** [comment]
:+1:

File: BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
+$(document).on('focusin', (e) ->
+  if ($(e.target).closest(".mce-window").length)
+    e.stopImmediatePropagation();
+)
+
 activateTinyMce = (view, textarea) ->
+  $.ajax
+    url: $("#contextual-informations").data("translation-url-pattern").replace("*domain*", "tinymce")
+    success: (response) ->
+      tinymce.util.I18n.add response.lcoale, response.catalog
```
Comment (nicolasThal) [inline BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee]: respone.locale

File: BackofficeBundle/Resources/views/layout.html.twig [hunk 2]
```diff
@@ -65,6 +65,8 @@
 
     <script data-pace-options='{ "restartOnRequestAfter": true }' src="{{ asset('js/pace.min.js') }} "></script>
 
+    {{ tinymce_init() }}
```
Comment (nicolasThal) [inline BackofficeBundle/Resources/views/layout.html.twig]: Why did you move the tinymce activation before all the js load ?

File: BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee [hunk 3]
```diff
@@ -1,7 +1,22 @@
 # ACTIVATE TINYMCE
+
 callback_tinymce_init = null
 
+doCallBack = (editor, view) ->
```
Comment (nicolasThal) [inline BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee]: What is the usage of these anonymous empty function ?

File: BackofficeBundle/Resources/views/layout.html.twig (hunk 2 above)
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/views/layout.html.twig]: @nicolasThal : In my first attempt I need this as some scripts in our all.js required tinymce to be loaded. I think it's no more the case, but as it changes nothing to be loaded before or after our all.js I suggest to let it before. As some of our future bundles should be related to tinymce, it's more logical to load tinymce before execute plugins.

File: BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee (hunk 3 above)
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee]: @nicolasThal : I dind't introduce that lambda function, I've just updated it. It's a legacy from a preview work of @itkg-rboessel and @itkg-nanne, so they are the best placed to answer this.
But what I have understood discussing with @itkg-nanne is that this mechanism is introduced to deal with a specific Orchestra combo: only one media modal for several rich textes on the same form. That way, each rich text can locally run that callback, and so the modal correctly knows its parent.

File: BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee (hunk 1 above)
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee]: @nicolasThal : It's weird that the code works with that anomaly Oo

File: BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee (hunk 3 above)
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee]: @nicolasThal : I have read your question too fast and did a half-answer. This callback is empty because it is called in this cms-bundle file for a job required in media-admin. So it does nothing in cms-bundle but it is overriden is media-admin.
I'm not fan of the way it works because it allows only one override for all the bundles, but as it was here before my work and as I have no alternative for the moment, I let this like that, but I suggest a refacto when we'll have a better way of doing this.

**Comment by itkg-ngilain** [comment]
@nicolasThal : update done, are you ok with this and my answers ?

File: BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee (hunk 3 above)
Comment (nicolasThal) [inline BackofficeBundle/Resources/public/coffee/tinyMCE/activateTinyMCE.coffee]: All right :).

This question was really only for my own knowledge, but I will agree that we open a discussion on a refacto on this part.

**Comment by nicolasThal** [comment]
:+1:
